File: apis/src/main/java/com/beat/apis/performance/application/dto/create/CastResponse.java
package com.beat.apis.performance.application.dto.create;

import com.beat.global.support.jackson.CdnImageUrl;

public record CastResponse(
	Long castId,
	String castName,
	String castRole,
	@CdnImageUrl String castPhoto
) {
	public static CastResponse of(
		Long castId,
		String castName,
		String castRole,
		String castPhoto
	) {
		return new CastResponse(
			castId,
			castName,
			castRole,
			castPhoto
		);
	}
}

File: apis/src/main/kotlin/com/beat/apis/performance/api/response/CastResponse.kt
package com.beat.apis.performance.api.response

import com.beat.application.frontoffice.performance.CastResult
import com.beat.apis.web.jackson.CdnImageUrl

@ConsistentCopyVisibility
data class CastResponse private constructor(
    val castId: Long?,
    val castName: String?,
    val castRole: String?,
    @field:CdnImageUrl val castPhoto: String?,
) {
    companion object {
        fun from(result: CastResult): CastResponse = CastResponse(result.id, result.name, result.role, result.photo)
    }
}
